src/screens/ProfileScreen/elements/KarmaScore.tsx

```diff
@@ -1,35 +1,101 @@
 import * as React from 'react';
 import {Text, View} from 'react-native';
 import FastImage from 'react-native-fast-image';
-import {colors} from '../../../utils/colors/index';
-import BetterScore from '../../../assets/better-score.png';
+import Tooltip from 'react-native-walkthrough-tooltip';
+import {TouchableWithoutFeedback} from 'react-native-gesture-handler';
+import BetterScore from '../../../assets/Logogram.png';
+import MemoIc_question_mark from '../../../assets/icons/Ic_question_mark';
+import {normalize} from '../../../utils/fonts';
+import {CircleGradient} from '../../../components/Karma/CircleGradient';
 
 type KarmaScoreProps = {
   score: number;
 };
 
 export const KarmaScore = ({score}: KarmaScoreProps) => {
+  const [isTooltipShown, setIsTooltipShown] = React.useState(false);
   return (
-    <View
-      style={{
-        backgroundColor: colors.darkBlue,
-        borderRadius: 32,
-        flexDirection: 'row',
-        justifyContent: 'center',
-        alignItems: 'center',
-        paddingHorizontal: 14,
-        paddingVertical: 5,
-        alignSelf: 'flex-start'
-      }}>
-      <FastImage
-        resizeMode={FastImage.resizeMode.contain}
-        source={BetterScore}
-        style={{width: 24, height: 24, marginRight: 8}}
-      />
-      <Text style={{fontWeight: '800', color: colors.white, fontSize: 21, marginRight: 8}}>
-        {score}
-      </Text>
-      <Text style={{fontSize: 12, fontWeight: '700', color: colors.white}}>Karma</Text>
-    </View>
+    <TouchableWithoutFeedback onPress={() => setIsTooltipShown(true)}>
+      <View
+        style={{
+          borderRadius: 32,
+          flexDirection: 'row',
+          justifyContent: 'center',
+          alignItems: 'center',
+          alignSelf: 'flex-start'
+        }}>
+        <View style={{marginRight: 8}}>
+          <CircleGradient size={normalize(28)} width={normalize(2.3)} fill={score}>
+            <FastImage
+              resizeMode={FastImage.resizeMode.center}
+              source={BetterScore}
+              style={{width: 28, height: 28}}
+            />
+          </CircleGradient>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `CircleGradient` component has been introduced with a dynamic fill based on the `karma` score. Ensure that the `karma` score passed as a prop is always a valid number to prevent potential rendering issues.



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
ignore